Brokerages/Fxcm/FxcmBrokerage.Messaging.cs

```diff
@@ -0,0 +1,443 @@
+
+            AutoResetEvent autoResetEvent;
+            lock (_locker)
+            {
+                _currentRequest = _gateway.requestTradingSessionStatus();
+                autoResetEvent = new AutoResetEvent(false);
+                _mapRequestsToAutoResetEvents[_currentRequest] = autoResetEvent;
+            }
+            if (!autoResetEvent.WaitOne(ResponseTimeout))
+                throw new TimeoutException(string.Format("FxcmBrokerage.LoadInstruments(): Operation took longer than {0} seconds.", (decimal)ResponseTimeout / 1000));
+        }
+
+        private void LoadAccounts()
+        {
+            AutoResetEvent autoResetEvent;
+            lock (_locker)
+            {
+                _currentRequest = _gateway.requestAccounts();
+                autoResetEvent = new AutoResetEvent(false);
+                _mapRequestsToAutoResetEvents[_currentRequest] = autoResetEvent;
+            }
+            if (!autoResetEvent.WaitOne(ResponseTimeout))
+                throw new TimeoutException(string.Format("FxcmBrokerage.LoadAccounts(): Operation took longer than {0} seconds.", (decimal)ResponseTimeout / 1000));
+
+            // Hedging MUST be disabled on all accounts
+            if (_accounts.Values.Any(account => account.getParties().getFXCMPositionMaintenance() != "N"))
+            {
+                throw new NotSupportedException("FxcmBrokerage.LoadAccounts(): The Lean engine does not support accounts with Hedging enabled. Please contact FXCM support to disable Hedging.");
+            }
+        }
+
+        private void LoadOpenOrders()
+        {
+            AutoResetEvent autoResetEvent;
+            lock (_locker)
+            {
+                _currentRequest = _gateway.requestOpenOrders();
+                autoResetEvent = new AutoResetEvent(false);
+                _mapRequestsToAutoResetEvents[_currentRequest] = autoResetEvent;
+            }
+            if (!autoResetEvent.WaitOne(ResponseTimeout))
+                throw new TimeoutException(string.Format("FxcmBrokerage.LoadOpenOrders(): Operation took longer than {0} seconds.", (decimal)ResponseTimeout / 1000));
+        }
+
+        private void LoadOpenPositions()
+        {
+            AutoResetEvent autoResetEvent;
+            lock (_locker)
+            {
+                _currentRequest = _gateway.requestOpenPositions();
+                autoResetEvent = new AutoResetEvent(false);
+                _mapRequestsToAutoResetEvents[_currentRequest] = autoResetEvent;
+            }
+            if (!autoResetEvent.WaitOne(ResponseTimeout))
+                throw new TimeoutException(string.Format("FxcmBrokerage.LoadOpenPositions(): Operation took longer than {0} seconds.", (decimal)ResponseTimeout / 1000));
+        }
+
+        /// <summary>
+        /// Gets the quotes for the symbol
+        /// </summary>
+        public List<MarketDataSnapshot> GetQuotes(List<string> symbols)
```

Comment:
From the code it looks like this is an FXCM specific symbol, USD/JPY, is this correct? If so, maybe we need to provide a distinction in parameter naming to highlight that there is a difference between fxcm and qc symbols.